WebAPI/Controllers/RolesAndPermissionsController.cs

```diff
@@ -17,6 +20,56 @@ public RolesAndPermissionsController(IRolesAndPermissionsManager rolesAndPermiss
             _rolesAndPermissionsManager = rolesAndPermissionsManager;
         }
 
+        /// <summary>
+        ///     Get All Available Roles
+        /// </summary>
+        /// <response code="200"> Available Roles</response>
+        /// <response code="403">UnAuthorized Access </response>
+        [ProducesResponseType(200, Type = typeof(IEnumerable<Role>))]
+        [ProducesResponseType(403, Type = null)]
+        [HttpGet("GetAllRoles")]
+        public IActionResult GetAllRoles()
+        {
+            return StatusCode(StatusCodes.Status201Created, _rolesAndPermissionsManager.GetAllRoles().ToList());
+        }
+
+        /// <summary>
+        ///     Gets the Role By Id
+        /// </summary>
+        /// <remarks>
+        ///     Id must be greater than 0
+        /// </remarks>
+        /// <param name="roleId">Role Id</param>
+        /// <response code="500">Role Id is not correct </response>
+        /// <response code="403">UnAuthorized Access </response>
+        /// <response code="200">Role</response>
+        [ProducesResponseType(200, Type = typeof(Role))]
+        [ProducesResponseType(403, Type = null)]
+        [ProducesResponseType(500, Type = null)]
+        [HttpGet("GetRoleById")]
+        public IActionResult GetRoleById(int roleId)
+        {
+            if (roleId <= 0) return BadRequest();
```

Comment:
you say you would return 500 if the id is wrong although it means internal server error which implies that it was the server's (not the user's) fault. However, you are returning 400 bad request anyway which is super confusing.
This is a recurring error in your code, please review the entire controller